plugin/src/main/java/org/opensearch/ml/rest/mcpserver/RestMcpConnectionMessageStreamingAction.java

```diff
@@ -122,31 +147,136 @@ protected void prepareRequestInternal(final String path, final String sessionId,
+                                                    log.error("Error occurred when handling message", e);
+                                                    channel.sendResponse(new BytesRestResponse(channel, new Exception(e)));
+                                                } catch (IOException ex) {
+                                                    log
+                                                        .error(
+                                                            "Failed to send exception response to client during message handling in local due to IOException"
+                                                        );
+                                                }
+                                            }))
+                                            .subscribe();
+                                    } else {
+                                        client
+                                            .execute(
+                                                MLMcpMessageAction.INSTANCE,
+                                                new MLMcpMessageRequest(sessionId, requestBody),
+                                                actionListener
+                                            );
+                                    }
+                                }
+                            } else {
+                                log.error("SessionId not found in cluster");
+                                try {
+                                    channel
+                                        .sendResponse(
+                                            new BytesRestResponse(
+                                                channel,
+                                                new IllegalArgumentException(
+                                                    "SessionId not found in cluster, please try to create session first"
+                                                )
+                                            )
+                                        );
+                                } catch (IOException ex) {
+                                    log.error("Failed to send exception response to client when session ID not found in cluster state");
+                                }
                             }
-                        }).onErrorResume(e -> Mono.fromRunnable(() -> {
+
+                        }, e -> {
                             try {
-                                channel.sendResponse(new BytesRestResponse(channel, new Exception(e)));
+                                channel.sendResponse(new BytesRestResponse(channel, e));
                             } catch (IOException ex) {
-                                log.error("Failed to send exception response to client during message handling due to IOException");
+                                log.error("Failed to get the session management index result with sessionId: {}", sessionId);
```

Comment:
Error message is misleading. Here the exception should be some failure when send exception response to client. I think the original exception make sense. 

This error message should be added after link 235.

Reply:
This is the error message when getIndex request failed, so we need to change the message here